go.mod

```diff
@@ -1,13 +1,100 @@
 module github.com/openshift/generic-admission-server
 
-go 1.13
+go 1.17
 
 require (
-	github.com/spf13/cobra v0.0.5
-	k8s.io/api v0.19.5
-	k8s.io/apimachinery v0.19.5
-	k8s.io/apiserver v0.19.0
-	k8s.io/client-go v0.19.5
-	k8s.io/component-base v0.19.5
-	k8s.io/klog/v2 v2.3.0
+	github.com/spf13/cobra v1.2.1
+	k8s.io/api v0.23.0
+	k8s.io/apimachinery v0.23.0
+	k8s.io/apiserver v0.23.0
+	k8s.io/client-go v0.23.0
+	k8s.io/component-base v0.23.0
+	k8s.io/klog/v2 v2.30.0
+)
+
+require (
+	github.com/fsnotify/fsnotify v1.4.9 // indirect
+	sigs.k8s.io/json v0.0.0-20211020170558-c049b76a60c6 // indirect
+	github.com/openshift/library-go v0.0.0-20210407140145-f831e911c638 // indirect
+)
+
+require (
```

Comment:
why these three require sections? Is that what `go mod tidy` creates?

Reply:
Yes, they are created by `go mod`, it's a new feature in go1.17,  it will list all indirect lib explicitly in the go.mod file.
The key word for this feature is "Pruned module graphs in  go 1.17 modules".

Reply:
Oh, I think I got what you mean.  Those sections were generated by go mod. The smaller `require` section represents the lib I am able to use but not using currently.

I will remove those dependencies.